app/service/user_filter.rb

```diff
@@ -0,0 +1,21 @@
+class UserFilter
+  def initialize(query:, group_id: nil)
+    @query = query
+    @group_id = group_id
+  end
+
+  def run
+    users - group_members
+  end
+
+  private
+
+  def users
+    @users ||= User.filter_by_email(@query).sort_by_email.limit(10)
+  end
+
+  def group_members
+    return [] unless @group_id
+    @group_members ||= User.joins(:groups_users).where('group_id = ?', @group_id)
```

Comment:
More of a question about the best way to run this query, it's a nitpicky comment and not a huge deal.